packages/s2-core/src/cell/table-col-cell.ts

```diff
@@ -1,13 +1,54 @@
 import { get } from 'lodash';
-import { EXTRA_FIELD } from '../common/constant';
-import { renderDetailTypeSortIcon } from '../utils/layout/add-detail-type-sort-icon';
-import { getEllipsisText, getTextPosition } from '../utils/text';
+import { Group } from '@antv/g-canvas';
 import { renderText } from '@/utils/g-renders';
 import { ColCell } from '@/cell/col-cell';
+import { KEY_GROUP_FROZEN_COL_RESIZE_AREA } from '@/common/constant';
+
+import { renderDetailTypeSortIcon } from '../utils/layout/add-detail-type-sort-icon';
+import { getEllipsisText, getTextPosition } from '../utils/text';
 
 export class TableColCell extends ColCell {
-  protected getStyle() {
-    return get(this, 'theme.colCell');
+  protected isFrozenCol() {
+    const { frozenColCount, frozenTrailingColCount } = this.spreadsheet.options;
+    const { colIndex } = this.meta;
+    const colLeafNodes = this.spreadsheet.facet.layoutResult.colLeafNodes;
+
+    return (
+      (frozenColCount > 0 && colIndex < frozenColCount) ||
+      (frozenTrailingColCount > 0 &&
+        colIndex >= colLeafNodes.length - frozenTrailingColCount)
```

Comment:
是否能将并列的条件转换为有语义的变量？
```suggestion
 const xxx = colIndex >= colLeafNodes.length - frozenTrailingColCount;
 return  (frozenTrailingColCount > 0 && xxx)
```
或者适当增加点儿注释说明满足的条件场景便于日后维护